src/node/routes/vscode.ts

```diff
@@ -0,0 +1,99 @@
+import * as crypto from "crypto"
+import { Router } from "express"
+import { promises as fs } from "fs"
+import * as path from "path"
+import { commit, rootPath, version } from "../constants"
+import { authenticated, ensureAuthenticated, redirect, replaceTemplates } from "../http"
+import { getMediaMime, pathToFsPath } from "../util"
+import { VscodeProvider } from "../vscode"
+
+export const router = Router()
+
+const vscode = new VscodeProvider()
+
+router.get("/", async (req, res) => {
+  if (!authenticated(req)) {
+    return redirect(req, res, "login", {
+      to: req.baseUrl || "/",
+    })
+  }
+
+  const [content, options] = await Promise.all([
+    await fs.readFile(path.join(rootPath, "src/browser/pages/vscode.html"), "utf8"),
+    vscode
+      .initialize(
+        {
+          args: req.args,
+          remoteAuthority: req.headers.host || "",
+        },
+        req.query,
```

Comment:
Getting this error here, any idea why that might be?
```
[tsc] src/node/routes/vscode.ts:29:9 - error TS2345: Argument of type 'ParsedQs' is not assignable to parameter of type 'Query'.        │
[tsc]   Index signatures are incompatible.                                                                                              │
[tsc]     Type 'string | string[] | ParsedQs | ParsedQs[] | undefined' is not assignable to type 'string | string[] | undefined'.       │
[tsc]       Type 'ParsedQs' is not assignable to type 'string | string[] | undefined'.                                                  │
[tsc]         Type 'ParsedQs' is not assignable to type 'string'.                                                                       │
[tsc] 29         req.query,                                                                                                             │
```

Reply:
You might need to re-apply the VS Code patch. It accepts the query and I had to modify it to match Express.